activerecord/lib/active_record/model_schema.rb

```diff
@@ -349,6 +349,9 @@ def attributes_builder # :nodoc:
       end
 
       def columns_hash # :nodoc:
+        unless table_exists?
+          raise ActiveRecord::ActiveRecordError, "columns_hash is not available for a record with no table"
```

Comment:
`for a record with no table` sounds weird, because this is not a record's method and a record surely doesn't own any table. Shouldn't it be `for a model with no table`?

Reply:
You know what, in my mind I used record to differentiate a persisted model from a non persisted model, which I just call model.

But seeing in the Rails docs that this is not a distinction that is made and that model tends to refer to an active record model, so I think your suggestion is more appropriate here and consistent, nice catch, thank you.